src/plugins/discover/public/utils/use_row_heights_options.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { EuiDataGridRowHeightOption, EuiDataGridRowHeightsOptions } from '@elastic/eui';
+import { useMemo } from 'react';
+import { IUiSettingsClient } from 'kibana/public';
+import { Storage } from '../../../kibana_utils/public';
+import { ROW_HEIGHT_OPTION } from '../../common';
+import { getDefaultRowHeight, updateStoredRowHeight } from './row_heights';
+
+interface UseRowHeightProps {
+  rowHeightState?: number;
+  onUpdateRowHeight?: (rowHeight: number) => void;
+  storage: Storage;
+  uiSettings: IUiSettingsClient;
+}
+
+const SINGLE_ROW_HEIGHT_OPTION = 0;
+const AUTO_ROW_HEIGHT_OPTION = -1;
```

Comment:
Think we should add some descriptive comment here to explain what we're doing :)